packages/core/src/codewhisperer/region/regionProfileManager.ts

```diff
@@ -367,4 +450,63 @@ export class RegionProfileManager {
 
         return c
     }
+
+    // Should reset cache when users signout in case users want to change to a different connection
+    async resetCache() {
+        await globals.globalState.update('aws.amazonq.regionProfiles.cachedResult', {
+            isAcquired: false,
+            result: undefined,
+        })
+    }
+
+    private async tryAcquireLock(): Promise<CachedApiResultWithLock | undefined> {
+        const cachedValue = globals.globalState.tryGet<CachedApiResultWithLock>(
+            'aws.amazonq.regionProfiles.cachedResult',
+            Object,
+            { isAcquired: false, result: undefined }
+        )
+
+        if (!cachedValue.isAcquired) {
+            await globals.globalState.update('aws.amazonq.regionProfiles.cachedResult', {
+                ...cachedValue,
+                isAcquired: true,
+            })
+
+            return cachedValue
+        }
+        return undefined
+    }
+
+    private async releaeLock(cached: CachedApiResultWithLock) {
+        await globals.globalState.update('aws.amazonq.regionProfiles.cachedResult', {
+            ...cached,
+            isAcquired: false,
+        })
+    }
+
+    private async updateCacheWithResult(r: RegionProfile[]) {
+        const result: SuccessResult = {
+            type: 'success',
+            profiles: r,
+            timestamp: now(),
+        }
+        const pojo: CachedApiResultWithLock = {
+            isAcquired: false, // release the lock
```

Comment:
would the name `locked` be more obvious?
```suggestion
            locked: false, // release the lock
```

Reply:
makes sense